plugins/bigquery/dbt/adapters/bigquery/connections.py

```diff
@@ -294,7 +294,14 @@ def drop_dataset(self, database, schema):
         client = conn.handle
 
         with self.exception_handler('drop dataset'):
-            for table in client.list_tables(dataset):
+            try:
```

Comment:
can we replace this with:
```
client.delete_dataset(dataset_id, delete_contents=True, not_found_ok=True)
```

If the version of `google-cloud-bigquery` that we use supports it? If not, we can tackle it in a subsequent PR / release.

via [the bq docs](https://googleapis.dev/python/bigquery/latest/generated/google.cloud.bigquery.client.Client.html#google.cloud.bigquery.client.Client.delete_dataset)

Reply:
It looks like we can set a minimum of `1.15.0` and be sure we have that flag available - it's hard to find any information about earlier versions.

Reply:
ok - let's make that change in a separate PR, I'll queue up an issue